app/Post_Types/Archive_Item/Rest.php

```diff
@@ -0,0 +1,343 @@
+			'meta_query' => $meta_query
+		);
+		$fields = [];
+		$posts_ids = get_posts($args);
+		foreach($posts_ids as $post_id) {
+			$fields[] = [
+				self::FIELD_INDEX_ID => $post_id,
+				self::FIELD_INDEX_TITLE => get_the_title($post_id),
+				self::FIELD_INDEX_TYPE => carbon_get_post_meta($post_id, FieldPostMeta::FIELD_TYPE, 'carbon_fields_container_field_variables'),
+				self::FIELD_INDEX_FIELD_ID => carbon_get_post_meta($post_id, FieldPostMeta::FIELD_ID, 'carbon_fields_container_field_variables'),
+				self::FIELD_INDEX_FIELD_POPUP => carbon_get_post_meta($post_id, FieldPostMeta::FIELD_BROWSE_DISPLAY, 'carbon_fields_container_field_variables'),
+			];
+		}
+		$this->fields = $fields;
+		return $this->fields;
+	}
+
+	public function decorate_taxonomy_args($field, $args, $request ) {
+		if (empty($request[$field[self::FIELD_INDEX_FIELD_ID]])) {
+			return $args;
+		}
+		return $args;
+	}
+
+
+	public function decorate_cpt_args($field, $args, $request ) {
+		if ( empty( $request[$field[self::FIELD_INDEX_FIELD_ID]] ) ) {
+			return $args;
+		}
+		$value = $request[$field[self::FIELD_INDEX_FIELD_ID]];
+
+		// error_log(print_r( $value, true ) , 3, "/Applications/MAMP/logs/php_error.log");
+
+		global $wp;
+		$vars = apply_filters( 'query_vars', $wp->public_query_vars );
+		if ( ! isset($args[ 'meta_query' ])) {
+			$args[ 'meta_query' ] = [
+				'relation'		=> 'AND',
+			];
+		}
+		$field_cpt_id = carbon_get_post_meta( $field[self::FIELD_INDEX_ID],FieldPostMeta::FIELD_CPT_ID );
+		$args[ 'meta_query' ][] = [
+			'key'		=> $field[self::FIELD_INDEX_FIELD_ID],
+			'value'		=> sprintf('post:%s:%s',
+				$field_cpt_id,
+				$value
+			),
+			'compare'	=> '='
+		];
+
+		return $args;
+	}
+
+	public function decorate_select_args($field, $args, $request ) {
+		if ( empty( $request[$field[self::FIELD_INDEX_FIELD_ID]] ) ) {
+			return $args;
+		}
+		$value = $request[$field[self::FIELD_INDEX_FIELD_ID]];
+
+		error_log(print_r( $value, true ) , 3, "/Applications/MAMP/logs/php_error.log");
+
+		global $wp;
+		$vars = apply_filters( 'query_vars', $wp->public_query_vars );
+		if ( ! isset($args[ 'meta_query' ])) {
+			$args[ 'meta_query' ] = [
+				'relation'		=> 'AND',
+			];
+		}
+		$args[ 'meta_query' ][] = [
+			'key'		=> $field[self::FIELD_INDEX_FIELD_ID],
+			'value'		=> $value,
+			'compare'	=> 'IN'
+		];
+
+		return $args;
+	}
+
+	public function decorate_date_args($field, $args, $request ) {
+		if (empty($request[$field[self::FIELD_INDEX_FIELD_ID]])) {
+			return $args;
+		}
+		$range = $request[$field[self::FIELD_INDEX_FIELD_ID]];
+
+		// assume the format is array of dates in format YYY/MM/DD
+		if ( ! isset($args[ 'meta_query' ])) {
+			$args[ 'meta_query' ] = [
+				'relation'		=> 'AND',
+			];
+		}
+
+		if (isset($range[0])) {
+			$start_time = strtotime($range[0]);
+			$start_date = date("Ymd", $start_time);
+			$args[ 'meta_query' ][] = [
+				'key'		=> $field[self::FIELD_INDEX_FIELD_ID],
+				'compare'	=> '>=',
```

Comment:
Add a `type` entry here and anywhere you know you are comparing to a specific type.
Do not assume the date format will always make sense in its string form.
See [the codex](https://codex.wordpress.org/Class_Reference/WP_Query#Custom_Field_Parameters) under the `type` entry; here you want, probably, to use `DATE` or `DATETIME`.

Reply:
done